docs/docs/use-vast/ingest/README.md

```diff
@@ -0,0 +1,652 @@
+perform adjustments, such as replacing some generic types with more semantic
+aliases, e.g., using the `timstamp` alias instead of type `time` to designate
+the event timestamp.
+
+## Write a schema manually
+
+A [schema][schemas] is a record [type][types] with a name so that VAST can
+represent it as a table internally. You would write a schema manually or extend
+an existing schema if your goal is tuning type semantics and performance. For
+example, if you have a field of type `string` that only holds IP addresses, you
+can upgrade it to type `addr` and enjoy the benefits of richer query
+expressions, e.g., top-k prefix search. Or if you onboard a new data source, you
+can ship a schema along with [concept][concepts] mappings for a deeper
+integration.
+
+You write a schema (and potentially accompanying types, concepts, and models) in
+[module][modules].
+
+Let's write one from scratch, for a tiny dummy data source called *foo* that
+produces CSV of this kind:
+
+```csv
+date,target,message
+2022-05-17,10.0.0.1,foo
+2022-05-18,10.0.0.2,bar
+2022-05-18,10.0.0.3,bar
+```
+
+The corresponding schema type looks like this:
+
+```yaml
+message:
+  record:
+    - date: time
+    - target: addr
+    - message: msg
+```
+
+You can embed this type definition in a dedicated `foo` module:
+
+```yaml
+module: foo
+types:
+  message:
+    record:
+      - date: time
+      - target: addr
+      - message: msg
+```
+
+Place this module in an existing module directory, such as`/etc/vast/schema`, or
+tell VAST in your `vast.yaml` configuration file where to look for additional
+modules via the `module-dirs` key:
+
+```yaml
+vast:
+  module-dirs:
+    - path/to/modules
+```
+
+Restart VAST and you're ready to ingest the CSV with richer typing:
+
+```bash
+vast import csv < foo.csv
+```
+
+:::note Why YAML?
+Why YAML and not X? The primary reason is that YAML is the most common
+configuration language, and VAST uses it also for its configuration. But isn't
+the nesting excessive? Yes, it can be painful, but the alternatives are not
+rosier either. TOML and JSON create a lot of churn, and VAST's previous DSL is
+unnecessary cognitive load for a user who wants to get things done.
+:::
+
+## Map events to schemas
+
+For some input formats, such as JSON and CSV, VAST requires an existing
+[schema][schemas] to find the corresponding type definition and use higher-level
+types.
+
+There exist two ways to tell VAST how to map events to schemas:
+
+1. **Field Matching**: by default, VAST checks every new record whether there
+   exists a corresponding schema where the record fields match. If found, VAST
+   automatically assigns the matching schema.
+
+   The `--type=PREFIX` option makes it possible to restrict the set of candidate
+   schemas to type names with a given prefix, in case there exist multiple
+   schemas with identical field names.
+
+   VAST permanently tracks imported event types. They do not need to be
+   specified again for consecutive imports.
```

Comment:
The second sentence should be removed, as we removed that feature again as it caused a fair bit of trouble. The server still tracks the incoming table slice's schemas, but it hands out only those to the client that it found in module files.